madara/crates/client/submit_tx/src/validation.rs

```diff
@@ -311,6 +312,39 @@ impl TransactionValidator {
     }
 }
 
+#[async_trait]
+impl SubmitL1HandlerTransaction for TransactionValidator {
+    async fn submit_l1_handler_transaction(
+        &self,
+        tx: L1HandlerTransactionWithFee,
+    ) -> Result<L1HandlerTransactionResult, SubmitTransactionError> {
+        let arrived_at = TxTimestamp::now();
+
+        let (api_tx, class) = tx.clone().into_blockifier(
+            self.backend.chain_config().chain_id.to_felt(),
+            self.backend.chain_config().latest_protocol_version,
+        )?;
+
+        let response = L1HandlerTransactionResult {
+            transaction_hash : api_tx.tx_hash().to_felt(),
+        };
+
+        let validated_txn = ValidatedTransaction {
+            transaction: mp_transactions::Transaction::L1Handler(tx.tx.clone()),
+            paid_fee_on_l1: Some(tx.paid_fee_on_l1),
+            contract_address: tx.tx.contract_address,
+            arrived_at: TxTimestamp::now(),
```

Comment:
you already created one above
```suggestion
            arrived_at,
```